pkg/testing/integration/program.go

```diff
@@ -1728,6 +1728,37 @@ func (pt *ProgramTester) prepareGoProject(projinfo *engine.Projinfo) error {
 		return err
 	}
 
+	// initialize a go.mod for dependency resolution if one doesn't exist
+	_, err = os.Stat(filepath.Join(cwd, "go.mod"))
+	if err != nil {
+		err = pt.runCommand("go-mod-init", []string{goBin, "mod", "init"}, cwd)
+		if err != nil {
+			return err
+		}
+	}
+
+	err = pt.runCommand("go-mod-tidy", []string{goBin, "mod", "tidy"}, cwd)
+	if err != nil {
+		return err
+	}
+
+	pulumiSdkPath := filepath.Join(gopath, "src", "github.com", "pulumi", "pulumi", "sdk")
+
+	// if we have a local pulumi enlistment, test against that
+	_, err = os.Stat(pulumiSdkPath)
+	if err == nil {
+		editStr := fmt.Sprintf("github.com/pulumi/pulumi/sdk=%s", pulumiSdkPath)
+		err = pt.runCommand("go-mod-edit", []string{goBin, "mod", "edit", "-replace", editStr}, cwd)
+		if err != nil {
+			return err
+		}
+	}
+
+	err = pt.runCommand("go-mod-download", []string{goBin, "mod", "download"}, cwd)
+	if err != nil {
+		return err
+	}
+
 	// We only need to run dep-ensure if there is a Gopkg.toml file
 	_, err = os.Stat(filepath.Join(cwd, "Gopkg.toml"))
```

Comment:
It feels like the code below will interact awkwardly with the new code in the case that there is a `Gopkg.toml` but not a `go.mod`.  We'll end up preparing in both ways?  Seems we should either drop support for Gopkg.toml (and deal with the break thaat causes for existing tests as we bring in these updates), or else switch on whether there is a `Gopkg.toml` or `go.mod` and perhaps fail if there is neither?

Reply:
I'll remove the dep path and we can deal with upgrading tests incrementally as this gets rolled out.